# House Price Prediction

In [1]:
import numpy as np
import pandas as pd
from sklearn.linear_model import LinearRegression

In [2]:
df = pd.read_csv('house.csv')

In [3]:
df

area  bedrooms  age   price
0  2600         3   20  550000
1  3000         4   15  565000
2  3200         3   18  610000
3  3600         3   30  595000
4  4000         5    8  760000

In [4]:
x = df[['area', 'bedrooms', 'age']]

In [5]:
x

area  bedrooms  age
0  2600         3   20
1  3000         4   15
2  3200         3   18
3  3600         3   30
4  4000         5    8

In [6]:
y = df[['price']]

In [7]:
y

price
0  550000
1  565000
2  610000
3  595000
4  760000

In [8]:
model = LinearRegression()

In [9]:
model.fit(x,y)

LinearRegression()

In [10]:
model.coef_

array([[   137.25, -26025.  ,  -6825.  ]])

In [11]:
model.intercept_

array([383725.])

Multiple regression for house price prediction is:

y = 137.25 * area - 26025 * bedrooms - 6825 * age + 383725

In [12]:
model.predict([[3000,3,40]])

array([[444400.]])

In [14]:
model.predict([[2500,4,5]])

array([[588625.]])

In [15]:
house_predict = model.predict(x)
house_predict

array([[526000.],
       [589000.],
       [622000.],
       [595000.],
       [748000.]])